### ISSUE TITLE
Mesure de la consommation électrique

### ISSUE THREAD

**Issue by GrmpCerber** [thibdct/programmateur-fil-pilote-wifi#7]
Petite idée d'amélioration : mesurer la consommation électrique du circuit du cumulus.

Autre amélioration possible, mais qui nécessiterai une évolution majeure du câblage, mesurer la consommation électrique des 7 lignes dédiées au chauffage (ce qui nécessiterai donc de brancher leurs phases en plus de leur FP.

**Comment by fab33** [comment]
Peu utile à mon avis. Une carte openenergymonitor (emontx ou shield) permet de faire cela. De plus pour parler de mon cas je n'ai que 3 départs de chauffage pour 5 radiateurs.

**Comment by Azema** [comment]
Salut @GrmpCerber, 

En tant qu'électricien professionnel, je conseille de ne pas brancher directement le cumulus sur le relais de la carte Remora. Il vaut mieux passer par un contacteur modulaire.

Pour la consommation, tu as déjà la puissance apparente remontée par la télé-information (PAPP). Sinon, pour permettre de remonter les consommations pour chaque zone, ainsi que pour le relais, je pense que le mieux serait de pouvoir renseigner la puissance de chacun. Ensuite, il serait possible de retourner la consommation instantanée.
Mais le Remora ne dispose pas de mémoire pour stocker des informations, du moins pas pour des écritures répétées.

**Comment by hallard** [comment]
Ah oui je confirme le relais de la carte c'est pas fait pour piloter les grosses charges genre cumulus, sinon il va pas aimer !!!!

**Comment by fab33** [comment]
D'une part il ne va pas aimer et d'autre part il faut tout de même s'assurer de la présence d'un disjoncteur adapté sur la ligne. Le contacteur me semble obligatoire. Il faudrait ajouter un schéma de câblage car j'ai l'impression que l'erreur est courante.

**Comment by Azema** [comment]
Allez, parce que vous êtes sympa, je vous ai fait un joli petit dessin :-)

![relaisremora](https://cloud.githubusercontent.com/assets/147217/15625469/d3bd1172-24a7-11e6-8071-f872c5f0f94d.png)

**Comment by GrmpCerber** [comment]
Bonjour à tous et merci de vos réponses !

Un peu de contexte : mon chauffe eau est un 1650W et mon installation de chauffage est électrique.
Donc utiliser la PAPP ne m'est pas d'un grand secours quand les chauffages sont en marche, d'où ma question sur une mesure de conso.
Concernant la carte openenergymonitor elle est relativement chère, se base sur des pinces qui fournissent une mesure grossière et ... elle m'oblige à "éventrer" mon tableau électrique pour glisser les pinces au milieu des cables.
Pour ma part je penchais plus vers un composant (ou un ensemble de composant) tel le [LTC2945](http://cds.linear.com/docs/en/datasheet/2945fb.pdf) qui permet de récupérer en I²C la conso d'un circuit ... 80V :( (je n'ai pas trouvé d'équivalent 250V et je n'arrive pas à trouver la puissance max tolérée par le LTC2945)

Concernant la présence d'un disjoncteur adapté (== dont la puissance max est inférieure à la capacité du circuit), je n'aurais pas fait sans ;)
Maintenant sur le circuit, 1650W=6.6A sous 250VAC et, si je ne me trompe pas ( @Azema ? ), 7.7A sous 220VAC (pire hypothèse).
Donc pour être en sécurité j'ajoute 10% ==> il me faut un circuit qui "coupe" à 8.5A.
Concernant le disjoncteur adapté ça va être difficile de trouver du 8.5A donc je pousse ma marge de sécurité à 10A.
La BOM pointe un relais [Omron](http://www.conrad.fr/ce/fr/product/503893/Relais-pour-circuits-imprims-Omron-G6D-1A-ASI-5DC-5-VDC-5-A-1-NO-T-1-pcs/?utm_source=google-search-product&utm_medium=comparateur&utm_campaign=503893&WT.mc_id=comparateur-gsp-503893&LGWCODE=503893;43857;390&gclid=CjwKEAjwg6W6BRDn6v__7vzN9QkSJAC9l9C3PRAhiMjyKMdZ8HqBF9BJZ8oLUpdnkEX7JNu56MmZBxoCv8Pw_wcB) dont le maximum est 5A, ce n'est donc pas un bon candidat.
D'après le schéma du Remora (et c'est peut être là la source d'erreur de lecture) la board supporte un autre type de relai : le [HLS8L-DC5V-S-C](http://www.seeedstudio.com/depot/images/opl/datasheet/315030000.pdf) qui lui tolère les 10A.

NB je suis embêté par le manque de marge de sécurité entre les 10A du disjoncteur et les 10A du relais. Ma seule "sécurité" est donc le wattage max de mon cumulus actuel...
@Azema @hallard Y-a-t-il une erreur dans mon raisonnement ? cette solution est-elle viable ?

**Comment by Azema** [comment]
As-tu déjà un contacteur jour/nuit dans ton tableau électrique, car tu peux le piloter avec le relais.
Pour le calibrage du disjoncteur, tu es bon avec un 10A.
Le Remora est la partie commande du système, pas la partie puissance. C'est pourquoi il faut passer par un contacteur. Cela permettra de ne pas fragiliser ton relais, car le cumulus va tirer dur et pendant un long moment.

**Comment by fab33** [comment]
le pinces de mesure de courant ne sont pas imprécise mais au contraire très précises après calibration (à peine 2% par rapport au compteur). On trouve des sketch qui font même la différence entre puissance active/réactive et qui font du monitoring continu et non pas échantillonnage. Concernant le prix, le shield pour UNO ne coute que 11 livres ce qui ne me semble pas excessif. En prenant les pinces dur aliexpress on doit s'en sortir à un prix correct.
Pour le reste, laisser 7A dans un relais 10A pendant plusieurs heures, c'est pas top. Enfin sur le schéma il faut rajouter un disjoncteur 2A pour protéger le circuit de commande de la bobine et non pas la reprendre sur le 16A

**Comment by Azema** [comment]
Effectivement, tu as raison @fab33, mais j'étais un peu flemmard ce matin, car c'est ce que je fais dans mes tableaux.

**Comment by fab33** [comment]
tip top. Voila un bon schéma de référence !
